plugin/advisor/mysql/advisor_index_pk_type.go

```diff
@@ -0,0 +1,281 @@
+				pkDataList = append(pkDataList, pds...)
+			}
+		}
+	}
+	for _, pd := range pkDataList {
+		tableColumn, columnType := v.prettyPrintColumn(pd.table, pd.column, pd.columnType)
+		v.adviceList = append(v.adviceList, advisor.Advice{
+			Status:  v.level,
+			Code:    advisor.IndexPKHasUnexpectedType,
+			Title:   v.title,
+			Content: fmt.Sprintf("Columns in primary key must be INT/BIGINT but %s is %s", tableColumn, columnType),
+			Line:    pd.line,
+		})
+	}
+	return in, false
+}
+
+// Leave implements the ast.Visitor interface.
+func (*indexPkTypeChecker) Leave(in ast.Node) (ast.Node, bool) {
+	return in, true
+}
+
+func (v *indexPkTypeChecker) addNewColumn(tableName string, line int, colDef *ast.ColumnDef) []pkData {
+	var pkDataList []pkData
+	for _, option := range colDef.Options {
+		if option.Tp == ast.ColumnOptionPrimaryKey {
+			tp := v.getIntOrBigIntStr(colDef.Tp)
+			if tp != "INT" && tp != "BIGINT" {
+				pkDataList = append(pkDataList, pkData{
+					table:      tableName,
+					column:     []string{colDef.Name.String()},
+					columnType: []string{tp},
+					line:       line,
+				})
+			}
+		}
+	}
+	v.tablesNewColumns.set(tableName, colDef.Name.String(), colDef)
+	return pkDataList
+}
+
+func (v *indexPkTypeChecker) changeColumn(tableName, oldColumnName string, line int, newColumnDef *ast.ColumnDef) []pkData {
+	var pkDataList []pkData
+	v.tablesNewColumns.delete(tableName, oldColumnName)
+	for _, option := range newColumnDef.Options {
+		if option.Tp == ast.ColumnOptionPrimaryKey {
+			tp := v.getIntOrBigIntStr(newColumnDef.Tp)
+			if tp != "INT" && tp != "BIGINT" {
+				pkDataList = append(pkDataList, pkData{
+					table:      tableName,
+					column:     []string{newColumnDef.Name.String()},
+					columnType: []string{tp},
+					line:       line,
+				})
+			}
+		}
+	}
+	v.tablesNewColumns.set(tableName, newColumnDef.Name.String(), newColumnDef)
+	return pkDataList
+}
+
+func (v *indexPkTypeChecker) addConstraint(tableName string, line int, constraint *ast.Constraint) []pkData {
+	var pkDataList []pkData
+	if constraint.Tp == ast.ConstraintPrimaryKey {
+		if len(constraint.Keys) >= 2 {
+			var columnNames []string
+			var columnTypes []string
+			var inValidTypes bool
+			for _, key := range constraint.Keys {
+				columnName := key.Column.Name.String()
+				columnType := v.getPKColumnType(tableName, columnName)
+				if columnType != "INT" && columnType != "BIGINT" {
+					inValidTypes = true
+				}
+				columnNames = append(columnNames, key.Column.Name.String())
+				columnTypes = append(columnTypes, columnType)
+			}
+			if inValidTypes {
+				pkDataList = append(pkDataList, pkData{
+					table:      tableName,
+					column:     columnNames,
+					columnType: columnTypes,
+					line:       line,
+				})
+			}
+			return pkDataList
+		}
+		columnName := constraint.Keys[0].Column.Name.String()
+		columnType := v.getPKColumnType(tableName, columnName)
+		if columnType != "INT" && columnType != "BIGINT" {
```

Comment:
If there is no column in v.tablesNewColumns and catalog, we'd better skip the check instead of an advice

Reply:
LGTM